actions/java-gradle-test/action.yaml

```diff
@@ -0,0 +1,88 @@
+name: "Test java gradle"
+description: "Run unit and optionally integration tests using gradle"
+
+inputs:
+  integration-tests:
+    description: "If true, the action is executing integration tests (Default is true)"
+    required: false
+    default: true
+  sonar-token:
+    description: "Token for sonarcloud."
+    required: false
+  signing-secret-key-ring:
+    description: "Key ring file for signing the sonatype publication."
+    required: false
+  signing-key-id:
+    description: "Key id for signing the sonatype publication."
+    required: false
+  signing-password:
+    description: "Password for signing the sonatype publication."
+    required: false
+  java-distribution:
+    description: "Java distribution to be installed. (Default is temurin)"
+    required: false
+    default: "temurin"
+  java-version:
+    description: "Java version to be installed. (Default is 11)"
+    required: false
+    default: "11"
+  gradle-version:
+    description: "Gradle version to be installed. (Default is 6.7.1)"
+    required: false
+    default: "6.7.1"
+  gradle-cache:
+    description: "Cache to use. Options are gradle, maven, sbt. (Default is '')"
+    required: false
+    default: ""
+  working-directory:
+    description: "Working directory of your Gradle artifacts. (Default is root directory)"
+    required: false
+    default: "./"
+runs:
+  using: "composite"
+  steps:    
+    - name: Check out repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    
+    - name: Set up Gradle with version ${{ inputs.gradle-version }}
+      uses: bakdata/ci-templates/actions/java-gradle-setup@feature/java-gradle
+      with:
+        java-distribution: ${{ inputs.java-distribution }}
+        java-version: ${{ inputs.java-version }}
+        gradle-version: ${{ inputs.gradle-version }}
+        gradle-cache: ${{ inputs.gradle-cache }}
+
+    - name: Run unit tests
+      run: ./gradlew check --info --stacktrace -PexcludeTests="**/*IntegrationTest*"
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+    
+    - name: Run integration tests
```

Comment:
Some food for thought: A cool feature that was introduced in Gradle 7.3 for handling integration tests:
https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html

Reply:
Interesting, but we decided on skipping integration tests completely in the action, since they are not required for our opensource projects